app/models/movie.rb

```diff
@@ -1,3 +1,33 @@
 class Movie < ApplicationRecord
   belongs_to :studio
+  has_many :movie_actors
+  has_many :actors, through: :movie_actors
+
+  validates :title, presence: true
+  validates :creation_year, presence: true, numericality: true
+  validates :genre, presence: true
+
+  def concat_movie_details
+    "#{title} - #{creation_year} - #{genre}"
+  end
+
+  def order_actors
+    actors.order(:age)
+  end
+
+  def average_age
+    ages = []
+    actors.each do |actor|
+      ages << actor.age
+    end
+    average = ages.sum / ages.length.to_f.round(1)
```

Comment:
This approach is using Ruby instead of AR. The main red flag for me is when I see `.each` inside a model method. 

If you're looking to find an average of a given column, try using `.average` instead. Ex: 

```ruby
actors.average(:age)
```
This method will look at the `actors` association present in this model, and perform an average calculation of the given attribute on that model. An Actor instance has an `age` property of numeric value, so it can average them together for us.